tests/integ/test_basic.py

```diff
@@ -36,6 +40,110 @@ def test_crud(engine):
     assert [same_user] == excinfo.value.objects
 
 
+def test_model_defaults(engine):
+    class ColumnDefaultsModel(BaseModel):
+        hash = Column(Integer, hash_key=True, default=12)
+        range = Column(Integer, range_key=True, default=24)
+        other = Column(Integer, default=48)
+    engine.bind(ColumnDefaultsModel)
+
+    obj = ColumnDefaultsModel()
+    assert obj.hash == 12
+    assert obj.range == 24
+    assert obj.other == 48
+
+    engine.save(obj)
+
+    same_obj = ColumnDefaultsModel(hash=12, range=24)
+    engine.load(same_obj)
+
+    assert same_obj.hash == 12
+    assert same_obj.range == 24
+    assert same_obj.other == 48
+
+
+def test_model_defaults_load(engine):
+    class ColumnDefaultLoadModel(BaseModel):
+        hash = Column(Integer, hash_key=True, default=12)
+        range = Column(Integer, range_key=True, default=24)
+        other = Column(Integer, default=48)
+    engine.bind(ColumnDefaultLoadModel)
+
+    obj = ColumnDefaultLoadModel(hash=333, range=333)
+    engine.save(obj)
+
+    same_obj = ColumnDefaultLoadModel(hash=obj.hash, range=obj.range)
+    engine.load(same_obj)
+
+    assert same_obj.hash == 333
+    assert same_obj.range == 333
+    assert same_obj.other == 48
+
+
+def test_model_default_func(engine):
```

Comment:
please add a test that uses `nonlocal` to increment an int on each call, and validate that the call count matches the number of times the default should have been used (guarding against excessive work from eg. the load/dump/tracking/reference machinery adding bad behavior).

Reply:
👍